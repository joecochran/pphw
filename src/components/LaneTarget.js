import React from 'react';

function LaneTarget(props) {
  return (
    <div className="c-lane-target">
      <div>Drop here</div>
    </div>
  )
}

export default LaneTarget;
