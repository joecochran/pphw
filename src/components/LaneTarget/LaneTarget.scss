.c-lane-target {
  background: rgba(255,255,255,0.80);
  border: 1px dashed #C1C4C7;
  height: 93px;
  font-family: 'Gilroy', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 20px;
  color: #2D3439;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 16px;
}
