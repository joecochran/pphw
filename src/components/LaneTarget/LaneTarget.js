import React from 'react';
import './LaneTarget.scss';

function LaneTarget(props) {
  return (
    <div 
      className="c-lane-target"
      onDragEnter={(event) => {event.preventDefault()}}
      onDragOver={(event) => {event.preventDefault()}}
      onDrop={props.handleLaneDrop}>
        <div>Drop here</div>
    </div>
  )
}

export default LaneTarget;
