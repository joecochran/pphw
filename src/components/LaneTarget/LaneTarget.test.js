import React from 'react';
import LaneTarget from './LaneTarget';
import { shallow } from 'enzyme';

describe('<LaneTarget>', () => {
  const wrapper = shallow(<LaneTarget />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot after drop', () => {
    wrapper.simulate('dragenter', {
      preventDefault: () => {

      }
    });
    wrapper.simulate('dragover', {
      preventDefault: () => {

      }
    });
    wrapper.simulate('drop', {
      preventDefault: () => {

      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
