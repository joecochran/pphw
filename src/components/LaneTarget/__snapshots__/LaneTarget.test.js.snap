// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaneTarget> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LaneTarget />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Drop here
      </div>,
      "className": "c-lane-target",
      "onDragEnter": [Function],
      "onDragOver": [Function],
      "onDrop": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Drop here",
      },
      "ref": null,
      "rendered": "Drop here",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Drop here
        </div>,
        "className": "c-lane-target",
        "onDragEnter": [Function],
        "onDragOver": [Function],
        "onDrop": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Drop here",
        },
        "ref": null,
        "rendered": "Drop here",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<LaneTarget> matches snapshot after drop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LaneTarget />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Drop here
      </div>,
      "className": "c-lane-target",
      "onDragEnter": [Function],
      "onDragOver": [Function],
      "onDrop": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Drop here",
      },
      "ref": null,
      "rendered": "Drop here",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Drop here
        </div>,
        "className": "c-lane-target",
        "onDragEnter": [Function],
        "onDragOver": [Function],
        "onDrop": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Drop here",
        },
        "ref": null,
        "rendered": "Drop here",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
